tests/discovery/test_openalex.py

```diff
@@ -124,3 +124,120 @@ def test_query_error(query_params):
     discoverer = OpenAlex()
     with pytest.raises(QueryError):
         next(discoverer.query(**query_params))
+
+
+def test_focuses_drive_discovery_false():
+    """Test that focuses with drive_discovery=False are skipped."""
+    discoverer = OpenAlex()
+
+    # This should return no results because the focus is skipped
+    results = list(
+        discoverer.query(
+            # Focuses should take precedence over other parameters
+            author="Yoshua Bengio",
+            institution="mila",
+            focuses=Focuses([Focus(type="author", name="Yoshua Bengio", score=1.0)]),
+            page=1,
+            per_page=10,
+        )
+    )
+    assert len(results) == 0
+
+
+def test_focuses_author_type():
+    """Test that focuses with type='author' query by author name."""
+    discoverer = OpenAlex()
+
+    focuses = Focuses(
+        [
+            Focus(
+                type="author",
+                name="Yoshua Bengio",
+                score=10.0,
+                drive_discovery=True,
+            )
+        ]
+    )
+
+    # Query with focuses should return the same results as direct author query
+    focus_results = list(
+        discoverer.query(
+            # Focuses should take precedence over other parameters
+            author="Unknown Author",
+            institution="mila",
+            focuses=focuses,
+            page=1,
+            per_page=10,
+            limit=100,
+        )
+    )
+
+    direct_results = list(
+        discoverer.query(
+            author="Yoshua Bengio", institution="mila", page=1, per_page=10, limit=100
+        )
+    )
+
+    # Both should return the same papers
+    focus_papers = [p.paper.title for p in focus_results]
+    direct_papers = [p.paper.title for p in direct_results]
+
+    assert focus_papers == direct_papers
+    assert len(focus_results) == 10
+
+    # All focus results should have a score of 0.0
+    for result in focus_results:
+        assert result.score == 0.0
```

Comment:
I'm not sure if scores should be added to author names as well or if they are only for institution focuses?

Reply:
I think it was an oversight on my part not to rescore with the author scores. They should have the author score, I believe.